### PR TITLE
adds 1.24 cluster dockershim removal ready blog

### PR DESCRIPTION
**Pull request by katcosgrove**
This adds a short blog with issues to be aware of before upgrading to v1.24 related to the removal of dockershim, specifically highlighting areas in the documentation that cluster operators must pay attention to. This is not intended to be evergreen content. See issue [30917](https://github.com/kubernetes/website/issues/30917)

### CODE REVIEW

File: content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+layout: blog
+title: "Is Your Cluster Ready for v1.24?"
+date: 2022-03-31
+slug: ready-for-dockershim-removal
+---
+
+**Author:** Kat Cosgrove (Pulumi)
+
+
+Way back in December of 2020, Kubernetes announced the [deprecation of Dockershim](/blog/2020/12/02/dont-panic-kubernetes-and-docker/). In Kubernetes, dockershim is a software shim that allows you to use the entire Docker engine as your container runtime within Kubernetes. In the upcoming v1.24 release, we are removing Dockershim - the delay between deprecation and removal in line with the project’s policy of supporting features for at least one year after deprecation. If you are a cluster operator, this guide includes the practical realities of what you need to know going into this release. Also, what you need to do to make sure your cluster doesn’t fall over!
```
Comment (sftim) [inline content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:11]: I'm (still) a bit wary of “we are removing Dockershim” when it's not Pulumi doing it. _I_ know what's meant; is there any appreciable risk of someone getting the wrong idea here?

File: content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+Regardless of whether you are rolling your own cluster or using a managed Kubernetes service from a cloud provider, you may need to [migrate telemetry or security agents that rely on Docker Engine](/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents/). 
+
+## I have a Docker dependency. What now?
+
+If your Kubernetes cluster does depend on Docker Engine and you intend to upgrade to Kubernetes v1.24 (which you should do for security), you will need to either change your container runtime from Docker Engine to something else, or use [cri-dockerd](https://github.com/Mirantis/cri-dockerd). Since [containerd](https://containerd.io/) is a graduated CNCF project and the runtime within Docker itself, it’s a safe bet. Fortunately, the Kubernetes project has already documented the process of [changing a node’s container runtime](/docs/tasks/administer-cluster/migrating-from-dockershim/change-runtime-containerd/), using containerd as the example. Instructions are similar for switching to one of the other supported runtimes.
```
Comment (sftim) [inline content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:23]: `suggestion
If your Kubernetes cluster does depend on Docker Engine and you intend to upgrade to Kubernetes v1.24 (which you should eventually do for security and similar reasons), you will need to either change your container runtime from Docker Engine to something else, or use [cri-dockerd](https://github.com/Mirantis/cri-dockerd). Since [containerd](https://containerd.io/) is a graduated CNCF project and the runtime within Docker itself, it’s a safe bet as an alternative container runtime. Fortunately, the Kubernetes project has already documented the process of [changing a node’s container runtime](/docs/tasks/administer-cluster/migrating-from-dockershim/change-runtime-containerd/), using containerd as the example. Instructions are similar for switching to one of the other supported runtimes.
```

File: content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+title: "Is Your Cluster Ready for v1.24?"
+date: 2022-03-31
+slug: ready-for-dockershim-removal
+---
+
+**Author:** Kat Cosgrove (Pulumi)
+
+
+Way back in December of 2020, Kubernetes announced the [deprecation of Dockershim](/blog/2020/12/02/dont-panic-kubernetes-and-docker/). In Kubernetes, dockershim is a software shim that allows you to use the entire Docker engine as your container runtime within Kubernetes. In the upcoming v1.24 release, we are removing Dockershim - the delay between deprecation and removal in line with the project’s policy of supporting features for at least one year after deprecation. If you are a cluster operator, this guide includes the practical realities of what you need to know going into this release. Also, what you need to do to make sure your cluster doesn’t fall over!
+
```
Comment (sftim) [inline content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:12]: > If you are a cluster operator, this guide includes the practical realities of what you need to know going into this release. 

This _isn't_ the release notes - I'm genuinely worried that someone will think that Kubernetes v1.24 is imminently due and that this is the entire upgrade guide.

**Review by sftim — COMMENTED** [review]
Thanks @katcosgrove 

A few comments - what do you think?

File: content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md (hunk 1 above)
Comment (katcosgrove) [reply content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:11]: It doesn't give that impression to me, but if it's at all a risk I don't mind just removing my company affiliation from the blog. I only put it there because other blog authors tend to, anyway. :)

File: content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md (hunk 3 above)
Comment (katcosgrove) [reply content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:12]: Reasonable -- I will rephrase this to avoid any confusion

File: content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+Regardless of whether you are rolling your own cluster or using a managed Kubernetes service from a cloud provider, you may need to [migrate telemetry or security agents that rely on Docker Engine](/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents/). 
+
+## I have a Docker dependency. What now?
+
+If your Kubernetes cluster does depend on Docker Engine and you intend to upgrade to Kubernetes v1.24 (which you should eventually do for security and similar reasons), you will need to either change your container runtime from Docker Engine to something else, or use [cri-dockerd](https://github.com/Mirantis/cri-dockerd). Since [containerd](https://containerd.io/) is a graduated CNCF project and the runtime within Docker itself, it’s a safe bet as an alternative container runtime. Fortunately, the Kubernetes project has already documented the process of [changing a node’s container runtime](/docs/tasks/administer-cluster/migrating-from-dockershim/change-runtime-containerd/), using containerd as the example. Instructions are similar for switching to one of the other supported runtimes.
```
Comment (nate-double-u) [inline content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:23]:
```suggestion
If your Kubernetes cluster depends on Docker Engine and you intend to upgrade to Kubernetes v1.24 (which you should eventually do for security and similar reasons), you will need to change your container runtime from Docker Engine to something else or use [cri-dockerd](https://github.com/Mirantis/cri-dockerd). Since [containerd](https://containerd.io/) is a graduated CNCF project and the runtime within Docker itself, it’s a safe bet as an alternative container runtime. Fortunately, the Kubernetes project has already documented the process of [changing a node’s container runtime](/docs/tasks/administer-cluster/migrating-from-dockershim/change-runtime-containerd/), using containerd as an example. Instructions are similar for switching to one of the other supported runtimes.
```

File: content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+## I want to upgrade Kubernetes, and I need to maintain compatibility with Docker as a runtime. What are my options?
+
+Fear not, you aren’t being left out in the cold and you don’t have to take the security risk of staying on an old version of Kubernetes. Mirantis and Docker have jointly released, and are maintaining, a replacement for dockershim. That replacement is called [cri-dockerd](https://github.com/Mirantis/cri-dockerd). If you do need to maintain compatibility with Docker as a runtime, install cri-dockerd following the instructions in the project’s documentation.
```
Comment (nate-double-u) [inline content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:27]:
```suggestion
Fear not, you aren’t being left out in the cold and you don’t have to take the security risk of staying on an old version of Kubernetes. Mirantis and Docker have jointly released, and are maintaining, a replacement for dockershim. That replacement is called [cri-dockerd](https://github.com/Mirantis/cri-dockerd). If you need to maintain Docker compatibility as a runtime, install cri-dockerd following the instructions in the project’s documentation.
```

File: content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+## Is that it?
+
+
+Yes. As long as you go into this release aware of the changes being made and the details of your own clusters, as well as communicate clearly with your development teams, it will be minimally dramatic. You may have some changes to make to your cluster, your application code, or some scripts, but all of these requirements are documented. Switching from using Docker Engine as your runtime to using containerd effectively means removing the middleman, since containerd is the runtime used by Docker itself. From a practical perspective, this removal is better both for you and for Kubernetes maintainers in the long-run. If you still have questions, please first check the [Dockershim Removal FAQ](/blog/2022/02/17/dockershim-faq/).
```
Comment (nate-double-u) [inline content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:32]:
```suggestion
Yes. As long as you go into this release aware of the changes being made and the details of your own clusters and communicate clearly with your development teams, it will be minimally dramatic. You may have some changes to make to your cluster, application code, or scripts, but all of these requirements are documented. Switching from using Docker Engine as your runtime to using containerd effectively means removing the middleman since containerd is the runtime used by Docker itself. From a practical perspective, this removal is better both for you and for Kubernetes maintainers in the long-run. If you still have questions, please first check the [Dockershim Removal FAQ](/blog/2022/02/17/dockershim-faq/).
```

**Review by nate-double-u — COMMENTED** [review]
I did a copy edit pass with a couple word choice/sentence structure suggestions. Keep what you like :)

File: content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md (hunk 5 above)
Comment (sftim) [reply content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:27]: > If you need to maintain Docker compatibility as a runtime

To me that implies that we're calling the reader a runtime! I prefer the original wording.

File: content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md (hunk 6 above)
Comment (sftim) [reply content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:32]: For clarity, maybe this:
```suggestion
Yes. As long as you go into this release aware of the changes being made and the details of your own clusters, and you make sure to communicate clearly with your development teams, it will be minimally dramatic. You may have some changes to make to your cluster, application code, or scripts, but all of these requirements are documented. Switching from using Docker Engine as your runtime to using containerd effectively means removing the middleman since containerd is the runtime used by Docker itself. From a practical perspective, this removal is better both for you and for Kubernetes maintainers in the long-run.

If you still have questions, please first check the [Dockershim Removal FAQ](/blog/2022/02/17/dockershim-faq/).
```

**Comment by sftim** [comment]
/hold
This early approval stands; I just don't want a stray /lgtm to trigger a merge before Kat gets a chance to incorporate more feedback. @katcosgrove you can remove this /hold any time you wish to.

/approve

File: content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+layout: blog
+title: "Is Your Cluster Ready for v1.24?"
+date: 2022-03-31
+slug: ready-for-dockershim-removal
+---
+
+**Author:** Kat Cosgrove
+
+
+Way back in December of 2020, Kubernetes announced the [deprecation of Dockershim](/blog/2020/12/02/dont-panic-kubernetes-and-docker/). In Kubernetes, dockershim is a software shim that allows you to use the entire Docker engine as your container runtime within Kubernetes. In the upcoming v1.24 release, we are removing Dockershim - the delay between deprecation and removal in line with the project’s policy of supporting features for at least one year after deprecation. If you are a cluster operator, this guide includes the practical realities of what you need to know going into this release. Also, what you need to do to make sure your cluster doesn’t fall over!
```
Comment (mickeyboxell) [inline content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:11]:
```suggestion
Way back in December of 2020, Kubernetes announced the [deprecation of Dockershim](/blog/2020/12/02/dont-panic-kubernetes-and-docker/). In Kubernetes, dockershim is a software shim that allows you to use the Docker engine as your container runtime within Kubernetes. In the upcoming v1.24 release, we are removing Dockershim - the delay between deprecation and removal in line with the [project’s policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/) of supporting features for at least one year after deprecation. If you are a cluster operator, this guide includes the practical realities of what you need to know going into this release and what you need to do to make sure your cluster doesn’t fall over!
```

File: content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md (hunk 5 above)
Comment (mickeyboxell) [reply content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:27]: Haha agreed.

File: content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md (hunk 6 above)
Comment (mickeyboxell) [reply content/en/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:32]: Do you think it makes sense to specifically call out containerd here? I would like to see us be less opinionated about which supported runtime we advocate and instead direct people to the list of supported options: https://kubernetes.io/docs/setup/production-environment/container-runtimes/.

**Comment by katcosgrove** [comment]
/unhold

I'm happy with the state of it now! The date on this blog is set to 31 March, but we can change it to earlier.

**Comment by nate-double-u** [comment]
> /unhold

> I'm happy with the state of it now! The date on this blog is set to 31 March, but we can change it to earlier.

In that case, we can let the scheduler do its thing. Thanks for the post!

/lgtm

**Comment by nate-double-u** [comment]
(if we do wanna change the date, we can open a quick PR to do that)

**Comment by sftim** [comment]
Thanks @katcosgrove 

Nice one.
